src/hal/src/command/raw.rs

```diff
@@ -81,11 +81,20 @@ pub enum Level {
     Secondary,
 }
 
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub struct CommandBufferInheritanceInfo<'a, B: Backend> {
+    pub subpass: Option<pass::Subpass<'a, B>>,
+    pub framebuffer: Option<&'a B::Framebuffer>,
+    pub occlusion_query_flags: Option<QueryControl>,
+    pub pipeline_statistics: PipelineStatistic,
+}
+
 /// A trait that describes all the operations that must be
 /// provided by a `Backend`'s command buffer.
 pub trait RawCommandBuffer<B: Backend>: Clone + Any + Send + Sync {
     /// Begins recording commands to a command buffer.
-    fn begin(&mut self, flags: CommandBufferFlags);
+    fn begin(&mut self, flags: CommandBufferFlags, inheritance_info: Option<CommandBufferInheritanceInfo<B>>);
```

Comment:
wouldn't it be easier to always pass the inheritance info (without `Option`) given a `Default` implementation on it that inherits nothing?